pydrive2/auth.py

```diff
@@ -396,38 +390,58 @@ def LoadCredentialsFile(self, credentials_file=None):
         :raises: InvalidConfigError, InvalidCredentialsError
         """
         if credentials_file is None:
-            self._default_storage = self._storages["file"]
+            self._default_storage = self._storage_registry["file"]
+            credentials_file = self.settings.get("save_credentials_file")
             if self._default_storage is None:
                 raise InvalidConfigError(
                     "Backend `file` is not configured, specify "
                     "credentials file to read in the settings "
                     "file or pass an explicit value"
                 )
         else:
-            self._default_storage = Storage(credentials_file)
+            self._default_storage = FileBackend()
 
         try:
-            self.credentials = self._default_storage.get()
+            auth_info = self.default_storage.read_credentials(credentials_file)
```

Comment:
I think storage should be per file name. Otherwise we have to keep providing `credentials_file` everywhere + we use the same lock for all `credential` files potentially